### PR TITLE
Dynamic delete

### PR DESCRIPTION
**Pull request by alannesta**
## Change logs:
*  add a scheduler for scheduling delete tasks ( runs daily deleting movies that has been failed for a predefined times. Task is disabled for now)
* cache failed movieId and failure count in mem for 5 days
* check cache before parsing a movie (if previously failed too many times, skip parsing)
* update logger-es module, better logging

### CODE REVIEW

File: config.js [hunk 1]
```diff
@@ -50,5 +50,10 @@ module.exports = {
 			oauthCallbackUrl: getWeiboCallbackUrl(),
 			oauthRedirectUrl: getWeiboRedirectUrl()
 		}
+	},
+	DYNAMIC_CLEANUP_CONFIG: {
+		CACHE_KEY_PREFIX: 'INVALID_MOVIE_ID-',
+		CACHE_EXPIRE_SEC: 5 * 24 * 60 * 60,	// 5 days
+		THRESHOLD: 3	// delete if failed count is greater than 3
```
Comment (alannesta) [inline config.js]: change to 5

File: package.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
   "description": "91 parser",
   "main": "index.js",
   "scripts": {
-    "test": "mocha ./test/*.spec.js",
+    "test": "mocha --timeout 10000 ./test/*.spec.js",
```
Comment (alannesta) [inline package.json]: longer timeout for testing scheduler

File: server.js [hunk 3]
```diff
@@ -17,12 +18,17 @@ const logger = require('./utils/logger');
...
+const path = require('path');
+
 // instantiate related services
 CrawlerService.init();
 const configService = new ConfigService();
 
 ServiceManager.init();
 
+// register scheduled jobs
+Scheduler.init(path.join(process.cwd(), './services/scheduler/jobs'));
```
Comment (alannesta) [inline server.js]: __dirname ?

File: services/scheduler/scheduler-service.js [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+			if (task.type === 'interval') {
+				clearInterval(task.handler);
+				task.handler = setInterval(task.taskFn, time);
+			}
+
+			if (task.type === 'timeout') {
+				//console.log('clear timeout: ', task.handler);
+				clearTimeout(task.handler);
+				// update handler
+				task.handler = setTimeout(task.taskFn, time);
```
Comment (scaoupgrade) [inline services/scheduler/scheduler-service.js]: time * 1000, this is really confusing...

**Pull request merged**